Update dependencies

mpgxvii opened the pull request:
- Update dependencies and remove unused repositories
- Updated `radar-schemas` to version before latest

blootsvoets commented on gradle.properties:
```diff
@@ -3,10 +3,10 @@ dockerComposeStopContainers=true
 kotlinVersion=1.4.10
 okhttp3Version=4.9.+
-radarJerseyVersion=0.4.3
+radarJerseyVersion=0.6.0
```
Has been updated to https://github.com/RADAR-base/radar-jersey/releases/tag/v0.6.2

blootsvoets commented on gradle/dependency-locks/integrationTestRuntimeOnlyDependenciesMetadata.lockfile:
```diff
@@ -3,25 +3,25 @@
-org.glassfish.grizzly:grizzly-http-server-monitoring:2.4.4
-org.glassfish.grizzly:grizzly-http-server:2.4.4
-org.glassfish.grizzly:grizzly-http:2.4.4
+org.glassfish.grizzly:grizzly-framework-monitoring:2.4.5-SNAPSHOT
+org.glassfish.grizzly:grizzly-framework:2.4.5-SNAPSHOT
+org.glassfish.grizzly:grizzly-http-monitoring:2.4.5-SNAPSHOT
+org.glassfish.grizzly:grizzly-http-server-monitoring:2.4.5-SNAPSHOT
+org.glassfish.grizzly:grizzly-http-server:2.4.5-SNAPSHOT
+org.glassfish.grizzly:grizzly-http:2.4.5-SNAPSHOT
```
This should probably be 3.0.0

blootsvoets commented on gradle/dependency-locks/integrationTestRuntimeOnlyDependenciesMetadata.lockfile:
```diff
@@ -3,25 +3,25 @@
-org.junit.platform:junit-platform-commons:1.7.0
-org.junit.platform:junit-platform-engine:1.7.0
-org.junit:junit-bom:5.7.0
+org.junit.jupiter:junit-jupiter-api:5.8.0-SNAPSHOT
+org.junit.jupiter:junit-jupiter-engine:5.8.0-SNAPSHOT
+org.junit.platform:junit-platform-commons:1.8.0-SNAPSHOT
+org.junit.platform:junit-platform-engine:1.8.0-SNAPSHOT
+org.junit:junit-bom:5.8.0-SNAPSHOT
```
This should probably be 5.7.2

blootsvoets commented on build.gradle.kts:
```diff
@@ -19,15 +19,11 @@ dependencyLocking {
     // For working with dev-branches
-    maven(url = "https://repo.thehyve.nl/content/repositories/snapshots")
+    maven(url = "https://oss.sonatype.org/content/repositories/snapshots")
     maven(url = "https://oss.jfrog.org/artifactory/libs-snapshot/")
```
Is this still used?

yatharthranjan reviewed (commented):
Looks good. As joris suggested, the snapshot versions should not be used. Probably removing the snapshot repositories should fix that.
Otherwise, you can specify exact versions instead of `+`.

blootsvoets commented:
As a reference: in radar-gateway I moved away from using dependency ranges with locking and switched to fixed versions. It was replaced by running `./gradlew dependencyUpdates` once in a while. Changes in https://github.com/RADAR-base/RADAR-Gateway/commit/8b729998d3f7b0002014fdc765f5ad33981c66ef.

Main changes: use plugin with `id("com.github.ben-manes.versions") version "0.38.0"`, with the following gradle block:

```kotlin
    fun isNonStable(version: String): Boolean {
        val stableKeyword = listOf("RELEASE", "FINAL", "GA").any { version.toUpperCase().contains(it) }
        val regex = "^[0-9,.v-]+(-r)?$".toRegex()
        val isStable = stableKeyword || regex.matches(version)
        return isStable.not()
    }

    tasks.named<DependencyUpdatesTask>("dependencyUpdates").configure {
        rejectVersionIf {
            isNonStable(candidate.version)
        }
    }
```

mpgxvii commented:
> As a reference: in radar-gateway I moved away from using dependency ranges with locking and switched to fixed versions. It was replaced by running `./gradlew dependencyUpdates` once in a while. Changes in [RADAR-base/RADAR-Gateway@8b72999](https://github.com/RADAR-base/RADAR-Gateway/commit/8b729998d3f7b0002014fdc765f5ad33981c66ef).
> 
> Main changes: use plugin with `id("com.github.ben-manes.versions") version "0.38.0"`, with the following gradle block:
> 
> ```kotlin
>     fun isNonStable(version: String): Boolean {
>         val stableKeyword = listOf("RELEASE", "FINAL", "GA").any { version.toUpperCase().contains(it) }
>         val regex = "^[0-9,.v-]+(-r)?$".toRegex()
>         val isStable = stableKeyword || regex.matches(version)
>         return isStable.not()
>     }
> 
>     tasks.named<DependencyUpdatesTask>("dependencyUpdates").configure {
>         rejectVersionIf {
>             isNonStable(candidate.version)
>         }
>     }
> ```

Thanks, I've made the changes.

blootsvoets commented on gradle.properties:
```diff
@@ -2,17 +2,17 @@ org.gradle.jvmargs=-Xmx3072m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF
+slf4jVersion=1.7.30
+logbackVersion=1.2.3
+grizzlyVersion=3.0.0
+lzfseVersion=0.1.1
```
Lzfse should not be necessary for this project

blootsvoets commented on gradle.properties:
```diff
@@ -2,17 +2,17 @@ org.gradle.jvmargs=-Xmx3072m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF
+logbackVersion=1.2.3
+grizzlyVersion=3.0.0
+lzfseVersion=0.1.1
+kafkaVersion=2.7.0
+confluentVersion=5.5.2
```
I don't think these versions match. Please check the confluent release notes which confluent version corresponds to which kafka version.

blootsvoets commented on gradle.properties:
```diff
@@ -2,17 +2,16 @@ org.gradle.jvmargs=-Xmx3072m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF
 dockerComposeStopContainers=true
 
 kotlinVersion=1.4.10
```
Fix build by updating to 1.5.0
